go/benchmarks/glide_benchmark_client.go

```diff
@@ -0,0 +1,54 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package main
+
+import (
+	"github.com/aws/glide-for-redis/go/glide/api"
+)
+
+type glideBenchmarkClient struct {
+	client api.GlideClient
+}
+
+func (glideBenchmarkClient *glideBenchmarkClient) connect(connectionSettings *connectionSettings) error {
+	if connectionSettings.clusterModeEnabled {
+		config := api.NewRedisClusterClientConfiguration().
+			WithAddress(&api.NodeAddress{Host: connectionSettings.host, Port: connectionSettings.port}).
+			WithUseTLS(connectionSettings.useTLS)
+		glideClient, err := api.NewRedisClusterClient(config)
+		if err != nil {
+			return err
+		}
+
+		glideBenchmarkClient.client = glideClient
+		return nil
+	} else {
+		config := api.NewRedisClientConfiguration().
+			WithAddress(&api.NodeAddress{Host: connectionSettings.host, Port: connectionSettings.port}).
+			WithUseTLS(connectionSettings.useTLS)
+		glideClient, err := api.NewRedisClient(config)
+		if err != nil {
+			return err
+		}
+
+		glideBenchmarkClient.client = glideClient
+		return nil
+	}
+}
+
+func (glideBenchmarkClient *glideBenchmarkClient) get(key string) (string, error) {
+	return glideBenchmarkClient.client.Get(key)
+}
+
+func (glideBenchmarkClient *glideBenchmarkClient) set(key string, value string) (string, error) {
+	return glideBenchmarkClient.client.Set(key, value)
+}
```

Comment:
Are those sync operations?

Reply:
yes

Reply:
Should be async though
The goal of the project is async client and we need to benchmark async implementation then
Is goredis async? If no, you have to wrap it with async stuff.

Reply:
goredis is synchronous. Looking at the Java benchmarks [here](https://github.com/Bit-Quill/glide-for-redis/blob/main/java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java#L282), all the async operations immediately call `get()` to wait on the result. Even though the operations themselves are asynchronous, doesn't the immediate call to `get()` essentially make the operations synchronous? In other words, the thread that initiated the async operation doesn't do anything while the operation is being completed, it just blocks on the result

Reply:
Good point. Java benchmark calls sync client in a sync manner - I forgot about it